Index: src/chrome/service/service_process.cc
===================================================================
--- src/chrome/service/service_process.cc	(revision 65071)
+++ src/chrome/service/service_process.cc	(working copy)
@@ -121,10 +121,12 @@
   values->GetBoolean(prefs::kRemotingHostEnabled, &remoting_host_enabled);
   remoting_host_enabled |= command_line.HasSwitch(switches::kEnableRemoting);
 
+#if defined(ENABLE_REMOTING)
   // Check if remoting host is already enabled.
   if (remoting_host_enabled) {
     StartChromotingHost();
   }
+#endif
 
   // Enable Cloud Print if needed. First check the command-line.
   bool cloud_print_proxy_enabled =
