Index: src/base/i18n/break_iterator.cc
===================================================================
--- src/base/i18n/break_iterator.cc	(revision 69489)
+++ src/base/i18n/break_iterator.cc	(working copy)
@@ -5,7 +5,6 @@
 #include "base/i18n/break_iterator.h"
 
 #include "base/logging.h"
-#include "unicode/ubrk.h"
 #include "unicode/uchar.h"
 #include "unicode/ustring.h"
 
Index: src/base/i18n/break_iterator.h
===================================================================
--- src/base/i18n/break_iterator.h	(revision 69489)
+++ src/base/i18n/break_iterator.h	(working copy)
@@ -8,6 +8,7 @@
 
 #include "base/basictypes.h"
 #include "base/string16.h"
+#include "unicode/ubrk.h"
 
 // The BreakIterator class iterates through the words and word breaks
 // in a UTF-16 string.
@@ -77,11 +78,7 @@
   string16 GetString() const;
 
  private:
-  // ICU iterator, avoiding ICU ubrk.h dependence.
-  // This is actually an ICU UBreakiterator* type, which turns out to be
-  // a typedef for a void* in the ICU headers. Using void* directly prevents
-  // callers from needing access to the ICU public headers directory.
-  void* iter_;
+  UBreakIterator* iter_;
 
   // The string we're iterating over.
   const string16* string_;
