Index: src/chrome/browser/app_menu_model.cc
===================================================================
--- src/chrome/browser/app_menu_model.cc	(revision 44190)
+++ src/chrome/browser/app_menu_model.cc	(working copy)
@@ -55,10 +55,8 @@
 
   AddSeparator();
   AddCheckItemWithStringId(IDC_SHOW_BOOKMARK_BAR, IDS_SHOW_BOOKMARK_BAR);
-  AddItemWithStringId(IDC_FULLSCREEN, IDS_FULLSCREEN);
-  AddSeparator();
-  AddItemWithStringId(IDC_SHOW_HISTORY, IDS_SHOW_HISTORY);
   AddItemWithStringId(IDC_SHOW_BOOKMARK_MANAGER, IDS_BOOKMARK_MANAGER);
+  AddItemWithStringId(IDC_SHOW_HISTORY, IDS_SHOW_HISTORY);
   AddItemWithStringId(IDC_SHOW_DOWNLOADS, IDS_SHOW_DOWNLOADS);
 
   // Create the manage extensions menu item.
Index: src/chrome/browser/views/accelerator_table_gtk.cc
===================================================================
--- src/chrome/browser/views/accelerator_table_gtk.cc	(revision 44190)
+++ src/chrome/browser/views/accelerator_table_gtk.cc	(working copy)
@@ -128,7 +128,7 @@
   { base::VKEY_F11, false, true, true, IDC_FULLSCREEN },
   { base::VKEY_DELETE, false, true, true, IDC_TASK_MANAGER },
   { base::VKEY_OEM_COMMA, false, true, false, IDC_SYSTEM_OPTIONS },
-  { base::VKEY_B, true, true, false, IDC_SHOW_BOOKMARK_BAR },
+  { base::VKEY_B, true, true, false, IDC_SHOW_BOOKMARK_MANAGER },
 #if !defined(OS_CHROMEOS)
   { base::VKEY_F1, false, false, false, IDC_HELP_PAGE },
 #endif
@@ -139,6 +139,7 @@
   { base::VKEY_C, true, true, false, IDC_COMPACT_NAVBAR },
   { base::VKEY_F, false, true, true, IDC_FULLSCREEN },
 #endif
+  { base::VKEY_E, true, true, false, IDC_MANAGE_EXTENSIONS},
 };
 
 const size_t kAcceleratorMapLength = arraysize(kAcceleratorMap);
Index: src/chrome/browser/gtk/accelerators_gtk.cc
===================================================================
--- src/chrome/browser/gtk/accelerators_gtk.cc	(revision 44190)
+++ src/chrome/browser/gtk/accelerators_gtk.cc	(working copy)
@@ -151,7 +151,7 @@
   { GDK_o, IDC_OPEN_FILE, GDK_CONTROL_MASK },
   { GDK_f, IDC_FIND, GDK_CONTROL_MASK },
   { GDK_p, IDC_PRINT, GDK_CONTROL_MASK },
-  { GDK_b, IDC_SHOW_BOOKMARK_BAR,
+  { GDK_b, IDC_SHOW_BOOKMARK_MANAGER,
     GdkModifierType(GDK_CONTROL_MASK | GDK_SHIFT_MASK) },
   { GDK_F11, IDC_FULLSCREEN, GdkModifierType(0) },
   { GDK_Delete, IDC_CLEAR_BROWSING_DATA,
@@ -166,6 +166,8 @@
   { GDK_s, IDC_SAVE_PAGE, GDK_CONTROL_MASK },
   { GDK_e, IDC_SHOW_PAGE_MENU, GDK_MOD1_MASK },
   { GDK_f, IDC_SHOW_APP_MENU, GDK_MOD1_MASK },
+  { GDK_e, IDC_MANAGE_EXTENSIONS,
+    GdkModifierType(GDK_CONTROL_MASK | GDK_SHIFT_MASK) },
 };
 
 }  // namespace
