Index: src/chrome/browser/ui/gtk/accelerators_gtk.cc
===================================================================
--- src/chrome/browser/ui/gtk/accelerators_gtk.cc	(revision 94868)
+++ src/chrome/browser/ui/gtk/accelerators_gtk.cc	(working copy)
@@ -171,6 +171,8 @@
   { GDK_s, IDC_SAVE_PAGE, GDK_CONTROL_MASK },
   { GDK_e, IDC_SHOW_APP_MENU, GDK_MOD1_MASK },
   { GDK_f, IDC_SHOW_APP_MENU, GDK_MOD1_MASK },
+  { GDK_e, IDC_MANAGE_EXTENSIONS,
+    GdkModifierType(GDK_CONTROL_MASK | GDK_SHIFT_MASK) },
 };
 
 }  // namespace
Index: src/chrome/browser/ui/views/accelerator_table_gtk.cc
===================================================================
--- src/chrome/browser/ui/views/accelerator_table_gtk.cc	(revision 94868)
+++ src/chrome/browser/ui/views/accelerator_table_gtk.cc	(working copy)
@@ -141,6 +141,7 @@
   { ui::VKEY_H,              false, true,  false, IDC_SHOW_HISTORY },
   { ui::VKEY_F,              false, false, true,  IDC_SHOW_APP_MENU},
   { ui::VKEY_E,              false, false, true,  IDC_SHOW_APP_MENU},
+  { ui::VKEY_E, true, true, false, IDC_MANAGE_EXTENSIONS},
   { ui::VKEY_ESCAPE,         false, false, false, IDC_STOP },
   { ui::VKEY_ESCAPE,         true,  false, false, IDC_TASK_MANAGER },
   { ui::VKEY_U,              false, true,  false, IDC_VIEW_SOURCE },
