Index: src/chrome/browser/sessions/tab_restore_service.cc
===================================================================
--- src/chrome/browser/sessions/tab_restore_service.cc	(revision 46068)
+++ src/chrome/browser/sessions/tab_restore_service.cc	(working copy)
@@ -42,7 +42,7 @@
 // TabRestoreService ----------------------------------------------------------
 
 // static
-const size_t TabRestoreService::kMaxEntries = 10;
+const size_t TabRestoreService::kMaxEntries = 30;
 
 // Identifier for commands written to file.
 // The ordering in the file is as follows:
Index: src/chrome/browser/resources/new_new_tab.css
===================================================================
--- src/chrome/browser/resources/new_new_tab.css	(revision 46068)
+++ src/chrome/browser/resources/new_new_tab.css	(working copy)
@@ -24,7 +24,6 @@
   background: url(chrome://theme/product_logo) no-repeat 0 6px;
   position: relative;
   margin: 0 auto;
-  width: 920px;
   min-height: 50px;
   -webkit-transition: width .15s;
 }
@@ -206,7 +205,7 @@
 
 #recently-closed {
   white-space: nowrap;
-  overflow-x: hidden;
+  overflow-x: scroll;
 }
 
 #recently-closed :not(h2) {
Index: src/chrome/browser/resources/new_new_tab.js
===================================================================
--- src/chrome/browser/resources/new_new_tab.js	(revision 46068)
+++ src/chrome/browser/resources/new_new_tab.js	(working copy)
@@ -129,8 +129,6 @@
     var el = createRecentItem(item);
     parentEl.insertBefore(el, navEl);
   });
-
-  layoutRecentlyClosed();
 }
 
 function createRecentItem(data) {
Index: src/chrome/browser/dom_ui/new_tab_ui.cc
===================================================================
--- src/chrome/browser/dom_ui/new_tab_ui.cc	(revision 46068)
+++ src/chrome/browser/dom_ui/new_tab_ui.cc	(working copy)
@@ -39,13 +39,6 @@
 #include "grit/generated_resources.h"
 
 namespace {
-
-// The number of recent bookmarks we show.
-const int kRecentBookmarks = 9;
-
-// The number of search URLs to show.
-const int kSearchURLs = 3;
-
 // Strings sent to the page via jstemplates used to set the direction of the
 // HTML document based on locale.
 const wchar_t kRTLHtmlTextDirection[] = L"rtl";
@@ -264,7 +257,7 @@
   ListValue list_value;
   std::set<std::wstring> unique_items;
   int added_count = 0;
-  const int max_count = 10;
+  const int max_count = 30;
 
   // We filter the list of recently closed to only show 'interesting' entries,
   // where an interesting entry is either a closed window or a closed tab
Index: src/chrome/browser/dom_ui/most_visited_handler.cc
===================================================================
--- src/chrome/browser/dom_ui/most_visited_handler.cc	(revision 46068)
+++ src/chrome/browser/dom_ui/most_visited_handler.cc	(working copy)
@@ -33,7 +33,7 @@
 namespace {
 
 // The number of most visited pages we show.
-const size_t kMostVisitedPages = 8;
+const size_t kMostVisitedPages = 21;
 
 // The number of days of history we consider for most visited entries.
 const int kMostVisitedScope = 90;
