Index: src/chrome/browser/sessions/tab_restore_service.cc
===================================================================
--- src/chrome/browser/sessions/tab_restore_service.cc	(revision 50928)
+++ src/chrome/browser/sessions/tab_restore_service.cc	(working copy)
@@ -42,7 +42,7 @@
 // TabRestoreService ----------------------------------------------------------
 
 // static
-const size_t TabRestoreService::kMaxEntries = 10;
+const size_t TabRestoreService::kMaxEntries = 30;
 
 // Identifier for commands written to file.
 // The ordering in the file is as follows:
Index: src/chrome/browser/resources/new_new_tab.css
===================================================================
--- src/chrome/browser/resources/new_new_tab.css	(revision 50928)
+++ src/chrome/browser/resources/new_new_tab.css	(working copy)
@@ -23,7 +23,6 @@
   min-height: 100%;
   padding-bottom: 25px; /* Make room for the bottom positioned footer. */
   position: relative;
-  width: 920px;
 }
 
 html[dir=rtl] #main {
@@ -197,7 +196,7 @@
 
 #recently-closed {
   white-space: nowrap;
-  overflow-x: hidden;
+  overflow-x: scroll;
 }
 
 #recently-closed > span > span  {
Index: src/chrome/browser/resources/new_new_tab.html
===================================================================
--- src/chrome/browser/resources/new_new_tab.html	(revision 50928)
+++ src/chrome/browser/resources/new_new_tab.html	(working copy)
@@ -157,7 +157,10 @@
     </div>
 
     <div id="recently-closed" class="section" section="RECENT">
-      <h2 i18n-content="recentlyclosed"></h2>
+      <h2>
+      <a href="chrome://history/"
+         i18n-content="history"></a>
+      </h2>
       <span class="nav"></span>
     </div>
 
@@ -195,9 +198,6 @@
   </div>
 
   <div id="footer">
-    <a href="chrome://history"><span class="link-color"
-                                     i18n-content="history"></span></a>
-    <hr>
     <a href="chrome://downloads"><span class="link-color"
                                        i18n-content="downloads"></span></a>
     <hr>
Index: src/chrome/browser/resources/ntp/most_visited.css
===================================================================
--- src/chrome/browser/resources/ntp/most_visited.css	(revision 50928)
+++ src/chrome/browser/resources/ntp/most_visited.css	(working copy)
@@ -3,7 +3,7 @@
 #most-visited {
   position: relative;
   padding: 0;
-  height: 366px;
+  height: 550px;
   margin-top: -10px;
   -webkit-user-select: none;
 }
@@ -241,7 +241,7 @@
 
 @media (max-width: 940px) {
   #most-visited {
-    height: 294px;
+    height: 440px;
   }
 
   .thumbnail,
Index: src/chrome/browser/resources/ntp/most_visited.js
===================================================================
--- src/chrome/browser/resources/ntp/most_visited.js	(revision 50928)
+++ src/chrome/browser/resources/ntp/most_visited.js	(working copy)
@@ -234,9 +234,9 @@
               '<div></div>' +
             '</div>' +
           '</a>';
-      this.element.innerHTML = Array(8 + 1).join(singleHtml);
+      this.element.innerHTML = Array(21 + 1).join(singleHtml);
       var children = this.element.children;
-      for (var i = 0; i < 8; i++) {
+      for (var i = 0; i < 21; i++) {
         children[i].id = 't' + i;
       }
     },
@@ -244,8 +244,8 @@
     getMostVisitedLayoutRects_: function() {
       var small = this.useSmallGrid;
 
-      var cols = 4;
-      var rows = 2;
+      var cols = 7;
+      var rows = 3;
       var marginWidth = 10;
       var marginHeight = 7;
       var borderWidth = 4;
@@ -277,7 +277,7 @@
       if (this.visible) {
         var rects = this.getMostVisitedLayoutRects_();
         var children = this.element.children;
-        for (var i = 0; i < 8; i++) {
+        for (var i = 0; i < 21; i++) {
           var t = children[i];
           t.style.left = rects[i].left + 'px';
           t.style.top = rects[i].top + 'px';
@@ -491,7 +491,7 @@
     set data(data) {
       // We append the class name with the "filler" so that we can style fillers
       // differently.
-      var maxItems = 8;
+      var maxItems = 21;
       data.length = Math.min(maxItems, data.length);
       var len = data.length;
       for (var i = len; i < maxItems; i++) {
Index: src/chrome/browser/resources/new_new_tab.js
===================================================================
--- src/chrome/browser/resources/new_new_tab.js	(revision 50928)
+++ src/chrome/browser/resources/new_new_tab.js	(working copy)
@@ -71,8 +71,6 @@
   recentItems.forEach(function(item) {
     parentEl.appendChild(createRecentItem(item));
   });
-
-  layoutRecentlyClosed();
 }
 
 function createRecentItem(data) {
Index: src/chrome/browser/resources/downloads.html
===================================================================
--- src/chrome/browser/resources/downloads.html	(revision 50928)
+++ src/chrome/browser/resources/downloads.html	(working copy)
@@ -166,7 +166,6 @@
   color: #080;
   max-width: 500px;
   white-space: nowrap;
-  overflow: hidden;
   text-overflow: ellipsis;
 }
 
Index: src/chrome/browser/dom_ui/new_tab_ui.cc
===================================================================
--- src/chrome/browser/dom_ui/new_tab_ui.cc	(revision 50928)
+++ src/chrome/browser/dom_ui/new_tab_ui.cc	(working copy)
@@ -39,13 +39,6 @@
 #include "grit/generated_resources.h"
 
 namespace {
-
-// The number of recent bookmarks we show.
-const int kRecentBookmarks = 9;
-
-// The number of search URLs to show.
-const int kSearchURLs = 3;
-
 // Strings sent to the page via jstemplates used to set the direction of the
 // HTML document based on locale.
 const wchar_t kRTLHtmlTextDirection[] = L"rtl";
@@ -233,7 +226,7 @@
   ListValue list_value;
   std::set<std::wstring> unique_items;
   int added_count = 0;
-  const int max_count = 10;
+  const int max_count = 30;
 
   // We filter the list of recently closed to only show 'interesting' entries,
   // where an interesting entry is either a closed window or a closed tab
Index: src/chrome/browser/dom_ui/most_visited_handler.cc
===================================================================
--- src/chrome/browser/dom_ui/most_visited_handler.cc	(revision 50928)
+++ src/chrome/browser/dom_ui/most_visited_handler.cc	(working copy)
@@ -36,7 +36,7 @@
 namespace {
 
 // The number of most visited pages we show.
-const size_t kMostVisitedPages = 8;
+const size_t kMostVisitedPages = 21;
 
 // The number of days of history we consider for most visited entries.
 const int kMostVisitedScope = 90;
