Index: src/chrome/browser/sessions/tab_restore_service.cc
===================================================================
--- src/chrome/browser/sessions/tab_restore_service.cc	(revision 58578)
+++ src/chrome/browser/sessions/tab_restore_service.cc	(working copy)
@@ -49,7 +49,7 @@
 // TabRestoreService ----------------------------------------------------------
 
 // static
-const size_t TabRestoreService::kMaxEntries = 10;
+const size_t TabRestoreService::kMaxEntries = 30;
 
 // Identifier for commands written to file.
 // The ordering in the file is as follows:
Index: src/chrome/browser/resources/new_new_tab.css
===================================================================
--- src/chrome/browser/resources/new_new_tab.css	(revision 58578)
+++ src/chrome/browser/resources/new_new_tab.css	(working copy)
@@ -25,7 +25,7 @@
 #main,
 .section,
 .maxiview {
-  width: 920px;
+  width: 1630px;
 }
 
 html[dir=rtl] #main {
@@ -407,7 +407,7 @@
   display: none;
   margin: 10px 0 30px;
   white-space: nowrap;
-  overflow-x: hidden;
+  overflow-x: scroll;
 }
 
 .section.hidden > * {
Index: src/chrome/browser/resources/new_new_tab.html
===================================================================
--- src/chrome/browser/resources/new_new_tab.html	(revision 58578)
+++ src/chrome/browser/resources/new_new_tab.html	(working copy)
@@ -163,7 +163,7 @@
         noexpand="true">
       <h2>
         <div class="back"></div>
-        <span i18n-content="recentlyclosed"></span>
+        <a href="chrome://history/"><span i18n-content="recentlyclosed"></a>
       </h2>
       <div class="miniview"></div>
     </div>
Index: src/chrome/browser/resources/ntp/most_visited.js
===================================================================
--- src/chrome/browser/resources/ntp/most_visited.js	(revision 58578)
+++ src/chrome/browser/resources/ntp/most_visited.js	(working copy)
@@ -235,9 +235,9 @@
               '<div></div>' +
             '</div>' +
           '</a>';
-      this.element.innerHTML = Array(8 + 1).join(singleHtml);
+      this.element.innerHTML = Array(21 + 1).join(singleHtml);
       var children = this.element.children;
-      for (var i = 0; i < 8; i++) {
+      for (var i = 0; i < 21; i++) {
         children[i].id = 't' + i;
       }
     },
@@ -245,8 +245,8 @@
     getMostVisitedLayoutRects_: function() {
       var small = this.useSmallGrid;
 
-      var cols = 4;
-      var rows = 2;
+      var cols = 7;
+      var rows = 3;
       var marginWidth = 10;
       var marginHeight = 7;
       var borderWidth = 4;
@@ -279,7 +279,7 @@
       if (this.visible) {
         var rects = this.getMostVisitedLayoutRects_();
         var children = this.element.children;
-        for (var i = 0; i < 8; i++) {
+        for (var i = 0; i < 21; i++) {
           var t = children[i];
           t.style.left = rects[i].left + 'px';
           t.style.top = rects[i].top + 'px';
@@ -493,7 +493,7 @@
     set data(data) {
       // We append the class name with the "filler" so that we can style fillers
       // differently.
-      var maxItems = 8;
+      var maxItems = 21;
       data.length = Math.min(maxItems, data.length);
       var len = data.length;
       for (var i = len; i < maxItems; i++) {
Index: src/chrome/browser/resources/ntp/most_visited.css
===================================================================
--- src/chrome/browser/resources/ntp/most_visited.css	(revision 58578)
+++ src/chrome/browser/resources/ntp/most_visited.css	(working copy)
@@ -2,7 +2,7 @@
 
 #most-visited-maxiview {
   position: relative;
-  height: 366px;
+  height: 550px;
   -webkit-user-select: none;
 }
 
@@ -239,7 +239,7 @@
 
 @media (max-width: 940px) {
   #most-visited-maxiview {
-    height: 294px;
+    height: 440px;
   }
 
   .thumbnail,
Index: src/chrome/browser/resources/downloads.html
===================================================================
--- src/chrome/browser/resources/downloads.html	(revision 58578)
+++ src/chrome/browser/resources/downloads.html	(working copy)
@@ -166,7 +166,6 @@
   color: #080;
   max-width: 500px;
   white-space: nowrap;
-  overflow: hidden;
   text-overflow: ellipsis;
 }
 
Index: src/chrome/browser/dom_ui/new_tab_ui.cc
===================================================================
--- src/chrome/browser/dom_ui/new_tab_ui.cc	(revision 58578)
+++ src/chrome/browser/dom_ui/new_tab_ui.cc	(working copy)
@@ -44,13 +44,6 @@
 #include "grit/theme_resources.h"
 
 namespace {
-
-// The number of recent bookmarks we show.
-const int kRecentBookmarks = 9;
-
-// The number of search URLs to show.
-const int kSearchURLs = 3;
-
 // Strings sent to the page via jstemplates used to set the direction of the
 // HTML document based on locale.
 const char kRTLHtmlTextDirection[] = "rtl";
@@ -223,7 +216,7 @@
   ListValue list_value;
   std::set<std::string> unique_items;
   int added_count = 0;
-  const int max_count = 10;
+  const int max_count = 30;
 
   // We filter the list of recently closed to only show 'interesting' entries,
   // where an interesting entry is either a closed window or a closed tab
Index: src/chrome/browser/history/top_sites.cc
===================================================================
--- src/chrome/browser/history/top_sites.cc	(revision 58578)
+++ src/chrome/browser/history/top_sites.cc	(working copy)
@@ -35,8 +35,8 @@
 namespace history {
 
 // How many top sites to store in the cache.
-static const size_t kTopSitesNumber = 20;
-static const size_t kTopSitesShown = 8;
+static const size_t kTopSitesNumber = 21;
+static const size_t kTopSitesShown = 21;
 static const int kDaysOfHistory = 90;
 // Time from startup to first HistoryService query.
 static const int64 kUpdateIntervalSecs = 15;
