Index: src/chrome/browser/page_menu_model.cc
===================================================================
--- src/chrome/browser/page_menu_model.cc	(revision 47364)
+++ src/chrome/browser/page_menu_model.cc	(working copy)
@@ -20,14 +20,6 @@
 }
 
 void PageMenuModel::Build() {
-#if !defined(OS_CHROMEOS)
-#if defined(OS_MACOSX)
-  AddItemWithStringId(IDC_CREATE_SHORTCUTS, IDS_CREATE_APPLICATION_MAC);
-#else
-  AddItemWithStringId(IDC_CREATE_SHORTCUTS, IDS_CREATE_SHORTCUTS);
-#endif
-  AddSeparator();
-#endif
   AddItemWithStringId(IDC_CUT, IDS_CUT);
   AddItemWithStringId(IDC_COPY, IDS_COPY);
   AddItemWithStringId(IDC_PASTE, IDS_PASTE);
@@ -45,12 +37,10 @@
                          encoding_menu_model_.get());
 
   AddSeparator();
-  devtools_menu_model_.reset(new DevToolsMenuModel(delegate()));
-  AddSubMenuWithStringId(IDC_DEVELOPER_MENU, IDS_DEVELOPER_MENU,
-                         devtools_menu_model_.get());
-
-  AddSeparator();
-  AddItemWithStringId(IDC_REPORT_BUG, IDS_REPORT_BUG);
+  AddItemWithStringId(IDC_VIEW_SOURCE, IDS_VIEW_SOURCE);
+  if (g_browser_process->have_inspector_files())
+    AddItemWithStringId(IDC_DEV_TOOLS_CONSOLE, IDS_DEV_TOOLS_CONSOLE);
+  AddItemWithStringId(IDC_TASK_MANAGER, IDS_TASK_MANAGER);
 }
 
 ////////////////////////////////////////////////////////////////////////////////
