Index: src/third_party/WebKit/Source/WebCore/page/SecurityPolicy.cpp
===================================================================
--- src/third_party/WebKit/Source/WebCore/page/SecurityPolicy.cpp	(revision 102542)
+++ src/third_party/WebKit/Source/WebCore/page/SecurityPolicy.cpp	(working copy)
@@ -50,6 +50,23 @@
 
 bool SecurityPolicy::shouldHideReferrer(const KURL& url, const String& referrer)
 {
+    // Hide referrer if the url is on the blacklist.
+    String h = url.host();
+    if (h.contains("photo.sina.com.cn", false)
+        || h.contains("blog.sina.com.cn", false)
+        || h.contains("photo.store.qq.com", false)
+        || h.contains("imgsrc.baidu.com", false)
+        || h.contains("image.club.china.com", false)
+        || h.contains("yupoo.com", false)
+        || h.contains("pp.sohu.com.cn", false)
+        || h.contains("hiphotos.baidu.com", false)
+        || h.contains("ph.126.net", false)
+        || h.contains("img.blog.arting365.com", false)
+        || h.contains("babytreeimg.com", false)
+        || h.contains("img.cnbeta.com", false)
+        )
+        return true;
+
     bool referrerIsSecureURL = protocolIs(referrer, "https");
     bool referrerIsWebURL = referrerIsSecureURL || protocolIs(referrer, "http");
 
