Index: third_party/WebKit/WebCore/page/SecurityOrigin.cpp
===================================================================
--- src/third_party/WebKit/WebCore/page/SecurityOrigin.cpp	(revision 54103)
+++ src/third_party/WebKit/WebCore/page/SecurityOrigin.cpp	(working copy)
@@ -483,6 +483,13 @@
 
 bool SecurityOrigin::shouldHideReferrer(const KURL& url, const String& referrer)
 {
+    // Hide referrer if the url is on the blacklist.
+    String h = url.host();
+    if (h.contains("photo.sina.com.cn", false)
+        || h.contains("blog.sina.com.cn", false)
+        )
+        return true;
+
     bool referrerIsSecureURL = protocolIs(referrer, "https");
     bool referrerIsWebURL = referrerIsSecureURL || protocolIs(referrer, "http");
