Index: src/third_party/WebKit/WebCore/page/SecurityOrigin.cpp
===================================================================
--- src/third_party/WebKit/WebCore/page/SecurityOrigin.cpp	(revision 66724)
+++ src/third_party/WebKit/WebCore/page/SecurityOrigin.cpp	(working copy)
@@ -465,6 +465,20 @@
 
 bool SecurityOrigin::shouldHideReferrer(const KURL& url, const String& referrer)
 {
+    // Hide referrer if the url is on the blacklist.
+    String h = url.host();
+    if (h.contains("photo.sina.com.cn", false)
+        || h.contains("blog.sina.com.cn", false)
+        || h.contains("photo.store.qq.com", false)
+        || h.contains("imgsrc.baidu.com", false)
+        || h.contains("image.club.china.com", false)
+        || h.contains("yupoo.com", false)
+        || h.contains("pp.sohu.com.cn", false)
+        || h.contains("hiphotos.baidu.com", false)
+        || h.contains("ph.126.net", false)
+        )
+        return true;
+
     bool referrerIsSecureURL = protocolIs(referrer, "https");
     bool referrerIsWebURL = referrerIsSecureURL || protocolIs(referrer, "http");
 
