Index: src/chrome/browser/extensions/extension_dom_ui.cc
===================================================================
--- src/chrome/browser/extensions/extension_dom_ui.cc	(revision 45618)
+++ src/chrome/browser/extensions/extension_dom_ui.cc	(working copy)
@@ -70,7 +70,7 @@
 
 ExtensionDOMUI::ExtensionDOMUI(TabContents* tab_contents)
     : DOMUI(tab_contents) {
-  should_hide_url_ = true;
+  should_hide_url_ = false;
   bindings_ = BindingsPolicy::EXTENSION;
 
   // For chrome:// overrides, some of the defaults are a little different.
