Index: src/chrome/browser/extensions/extension_web_ui.cc
===================================================================
--- src/chrome/browser/extensions/extension_web_ui.cc	(revision 86235)
+++ src/chrome/browser/extensions/extension_web_ui.cc	(working copy)
@@ -136,7 +136,7 @@
   DCHECK(extension);
   // Only hide the url for internal pages (e.g. chrome-extension or packaged
   // component apps like bookmark manager.
-  should_hide_url_ = !extension->is_hosted_app();
+  should_hide_url_ = false;
 
   // The base class defaults to enabling web ui bindings, but we don't need
   // those.
