Index: src/chrome/browser/extensions/extension_dom_ui.cc
===================================================================
--- src/chrome/browser/extensions/extension_dom_ui.cc	(revision 61262)
+++ src/chrome/browser/extensions/extension_dom_ui.cc	(working copy)
@@ -131,7 +131,7 @@
   DCHECK(extension);
   // Only hide the url for internal pages (e.g. chrome-extension or packaged
   // component apps like bookmark manager.
-  should_hide_url_ = !extension->is_hosted_app();
+  should_hide_url_ = false;
 
   bindings_ = BindingsPolicy::EXTENSION;
   // Bind externalHost to Extension DOMUI loaded in Chrome Frame.
