Index: src/chrome/browser/ui/gtk/gtk_util.cc
===================================================================
--- src/chrome/browser/ui/gtk/gtk_util.cc	(revision 86235)
+++ src/chrome/browser/ui/gtk/gtk_util.cc	(working copy)
@@ -343,6 +343,8 @@
 }
 
 void CenterOverWindow(GtkWindow* window, GtkWindow* parent) {
+  // Do nothing for XMonad
+  return;
   gfx::Rect frame_bounds = gtk_util::GetWidgetScreenBounds(GTK_WIDGET(parent));
   gfx::Point origin = frame_bounds.origin();
   gfx::Size size = gtk_util::GetWidgetSize(GTK_WIDGET(window));
